-1

**Failed Tests**:   Build  HeaderConsistency  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17f73d/17019/summary.html
**COMMIT**: 591cfe3e0db321c6d974e7dd73612ec1cf236588
**CMSSW**: CMSSW_12_0_X_2021-07-19-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34563/17019/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @makortel cms-sw/cmssw#34532
 - @civanch cms-sw/cmssw#34497
 - @bsunanda cms-sw/cmssw#34539
 - @kpedro88 cms-sw/cmssw#34508

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17f73d/17019/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17f73d/17019/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Leaving Package HeterogeneousCore/SonicTriton
>> Package HeterogeneousCore/SonicTriton built
Entering library rule at src/HeterogeneousCore/SonicTriton/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc:3:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/interface/TritonService.h:13:10: fatal error: grpc_client.h: No such file or directory
   13 | #include "grpc_client.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc:3:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/interface/TritonService.h:13:10: fatal error: grpc_client.h: No such file or directory
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicCore/test/SonicDummyOneAnalyzer.cc
>> Entering Package HeterogeneousCore/SonicTriton
Entering library rule at src/HeterogeneousCore/SonicTriton/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc:3:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/interface/TritonService.h:13:10: fatal error: 'grpc_client.h' file not found
#include "grpc_client.h"
         ^~~~~~~~~~~~~~~
1 error generated.
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/plugins/TritonService.cc:3:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-19-1100/src/HeterogeneousCore/SonicTriton/interface/TritonService.h:13:10: fatal error: 'grpc_client.h' file not found
</pre>

--- a/FWCore/ParameterSet/scripts/edmConfigDump
+++ b/FWCore/ParameterSet/scripts/edmConfigDump
@@ -21,13 +21,12 @@ parser.add_argument("--pruneVerbose", action="store_true",
 options = parser.parse_args()
 handle = open(options.filename, 'r')
 loader=SourceFileLoader("pycfg", options.filename)
-loaded = types.ModuleType("pycfg",loader.name)
-loader.exec_module(loaded)
-#cfo = imp.load_source("pycfg", options.filename, handle)
-cmsProcess = cfo.process
-handle.close()
+mod=types.ModuleType(loader.name)
+loader.exec_module(mod)
+cmsProcess= mod.process
 
 if options.prune:
     cmsProcess.prune(options.pruneVerbose)
 
 print(cmsProcess.dumpPython())
+

--- a/HLTrigger/Configuration/python/Utilities.py
+++ b/HLTrigger/Configuration/python/Utilities.py
@@ -1,4 +1,4 @@
-import imp as _imp
+import types
 
 import HLTrigger.Configuration.Tools.options as _options
 import HLTrigger.Configuration.Tools.confdb  as _confdb
@@ -26,7 +26,7 @@ def loadHltConfiguration(process, menu, **args):
   args['fragment'] = True
   options = _build_options(**args)
 
-  hlt = _imp.new_module('hlt')
+  hlt = types.ModuleType('hlt')
   exec(_confdb.HLTProcess(options).dump(), globals(), hlt.__dict__)
   process.extend( hlt )
 

--- a/HLTrigger/Configuration/test/getDatasets.py
+++ b/HLTrigger/Configuration/test/getDatasets.py
@@ -2,7 +2,7 @@
 
 import sys
 import subprocess
-import imp
+import types
 import re
 import FWCore.ParameterSet.Config as cms
 
@@ -18,7 +18,7 @@ def extractDatasets(version, database, config):
   (out, err) = proc.communicate()
 
   # load the streams and Datasets
-  hlt = imp.new_module('hlt')
+  hlt = types.ModuleType('hlt')
   exec(out, globals(), hlt.__dict__)
 
   return hlt.process

